lib/project_types/extension/features/argo/base.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+require 'base64'
+require 'shopify_cli'
+require 'semantic/semantic'
+
+module Extension
+  module Features
+    module Argo
+      class Base
+        include SmartProperties
+
+        SCRIPT_PATH = %w(build main.js).freeze
+
+        NPM_LIST_COMMAND = %w(list).freeze
+        YARN_LIST_COMMAND = %w(list --pattern).freeze
+        NPM_LIST_PARAMETERS = %w(--prod).freeze
+        YARN_LIST_PARAMETERS = %w(--production).freeze
+
+        YARN_INSTALL_COMMAND = %w(install).freeze
+        YARN_INSTALL_PARAMETERS = %w(--silent).freeze
+        YARN_RUN_COMMAND = %w(run).freeze
+        YARN_RUN_SCRIPT_NAME = %w(build).freeze
+
+        def create(directory_name, identifier, context)
+          setup.call(directory_name, identifier, context)
+        end
+
+        def config(context)
+          js_system = ShopifyCli::JsSystem.new(ctx: context)
+          if js_system.package_manager == 'yarn'
+            run_yarn_install(context, js_system)
+            run_yarn_run_script(context, js_system)
+          end
+          filepath = File.join(context.root, SCRIPT_PATH)
+          context.abort(context.message('features.argo.missing_file_error')) unless File.exist?(filepath)
+          begin
+            {
+              renderer_version: extract_argo_renderer_version(context),
+              serialized_script: Base64.strict_encode64(File.open(filepath).read.chomp),
+            }
+          rescue StandardError
+            context.abort(context.message('features.argo.script_prepare_error'))
+          end
+        end
+
+        def setup
+          raise NotImplementedError, "'#{__method__}' must be implemented for #{self.class}"
+        end
+
+        def renderer_package
+          raise NotImplementedError, "'#{__method__}' must be implemented for #{self.class}"
```

Comment:
Since we are doing pattern matching using the `renderer_package`, it would be nice to have some kind of documentation for anyone extending this class. What are you thoughts?

Reply:
I thought the code itself was clear but I might be wrong. Perhaps it's clear to me because I'm the author so I added a line of comment to the method. Please let me know if it works. Thank you for your reviews and comments!

Reply:
Just making sure that there is no "false matching" while using renderer package as regex :) Developer extending the `Base` class, should provide uniquely identifiable renderer package name.

Reply:
If we are adding a comment/docs here, I think it should probably go above the method name.

Reply:
Thank you for adding the documentation :)